shapely/decorators.py

```diff
@@ -88,3 +90,44 @@ def wrapped(*args, **kwargs):
                 arr.flags.writeable = old_flag
 
     return wrapped
+
+
+def deprecate_positional(should_be_kwargs, category=DeprecationWarning):
```

Comment:
Nice! 

Not for this PR, but I was thinking that we could also deprecate passing the first argument as a keyword argument, i.e. passing it _not_ positional.  
We are quite consistent using `geometry` for functions taking a single geometry/array (but I see we also have some use `geometries`), and here the keyword is quite descriptive (although still unnecessary I would say). But for example I don't think anyone should do `shapely.intersection(a=.., b=..)`. I don't know if it is worth deprecating, but if I would write those APIs now again, I think I would use positional-only keywords for those.